### ISSUE TITLE
Add dimension reduction options

### ISSUE THREAD

**Issue by DM-Berger** [stfxecutables/df-analyze#12]
~~Currently PCA and kernel PCA are supported (the latter very poorly, as this seems to require careful tuning / kernel selection to actually work properly)~~ (EDIT: PCA is currently no longer available as an option). It might make sense to add:

- reduction via [UMAP](https://umap-learn.readthedocs.io/en/latest/)
- sklearn [FeatureAgglomeration](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.FeatureAgglomeration.html#sklearn.cluster.FeatureAgglomeration) or [RandomProjection](https://scikit-learn.org/stable/modules/random_projection.html#random-projection)
- PCA or UMAP on clusters derived from [HDBSCAN](https://scikit-learn.org/stable/modules/generated/sklearn.cluster.HDBSCAN.html) with correlation as the distance metric
